third_party/move/move-core/types/src/parser.rs

```diff
@@ -267,6 +267,20 @@ impl<I: Iterator<Item = Token>> Parser<I> {
         })
     }
 
+    fn parse_module_id(&mut self) -> Result<ModuleId> {
+        Ok(match self.next()? {
+            Token::Address(addr) => match self.next()? {
+                Token::Name(s) => ModuleId::new(AccountAddress::from_hex_literal(&addr)?, Identifier::new(s)?),
```

Comment:
So you allow `Token::Address` followed by `Token::Name` with nothing in between?  That seems odd.